### PR TITLE
CDRIVER-3091 add VERSIONINFO to dlls

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary

Add VERSIONINFO resource to bson-1.0.dll and mongoc-1.0.dll.

# Background & Motivation

The VERSIONINFO resource enables checking version information by inspecting the .dll file.

The version info can be viewed from Explorer:

| Before | After|
|--------|--------|
| before | after| 

Powershell can be used to inspect. This notably includes the `Debug` flag for debug builds:
```powershell

> Get-ItemProperty .\cmake-build\src\libbson\Debug\bson-1.0.dll VersionInfo

VersionInfo  : File:             C:\cygwin\home\Administrator\code\tasks\mongo-c-driver-C3091\cmake-build\src\libbson\Debug\bson-1.0.dll
               InternalName:     bson-1.0
               OriginalFilename: bson-1.0.dll
               FileVersion:      1.25.0-pre
               FileDescription:  A BSON Library for C
               Product:          MongoDB C Driver
               ProductVersion:   1.25.0-pre
               Debug:            True
               Patched:          False
               PreRelease:       False
               PrivateBuild:     False
               SpecialBuild:     False
               Language:         English (United States)
```

## Applying to static libraries

The resource file is only applied to the shared libraries (.dll) and not the static libraries (.lib). This PR assumes the resource file is not applicable to the static libraries.

Applying the `.rc` file to the static target did not change the output of `Get-ItemProperty .\cmake-build\src\libbson\Debug\bson-static-1.0.lib VersionInfo`.

libcurl similarly only applies `libcurl.rc` [to the shared library](https://github.com/curl/curl/blob/913eacf7730429f3de5d662691154ceb2aee8aa5/lib/CMakeLists.txt#L180).

### CODE REVIEW

File: src/libbson/CMakeLists.txt [hunk 1]
```diff
@@ -230,6 +230,9 @@ endif()
 if(ENABLE_SHARED)
    add_library(bson_shared SHARED)
    target_link_libraries(bson_shared PRIVATE $<BUILD_INTERFACE:bson_obj_pic>)
+   if (WIN32)
```
Comment (rcsanchez97) [inline src/libbson/CMakeLists.txt:233]:
```suggestion
   if(WIN32)
```

File: src/libmongoc/CMakeLists.txt [hunk 2]
```diff
@@ -813,6 +813,9 @@ if (MONGOC_ENABLE_STATIC_BUILD)
 endif ()
 
 add_library (mongoc_shared SHARED ${SOURCES} ${HEADERS} ${HEADERS_FORWARDING})
+if (WIN32)
```
Comment (rcsanchez97) [inline src/libmongoc/CMakeLists.txt:816]:
```suggestion
if(WIN32)
```

**Review by rcsanchez97 — APPROVED** [review]
LGTM

File: src/libbson/CMakeLists.txt [hunk 3]
```diff
@@ -230,6 +230,9 @@ endif()
 if(ENABLE_SHARED)
    add_library(bson_shared SHARED)
    target_link_libraries(bson_shared PRIVATE $<BUILD_INTERFACE:bson_obj_pic>)
+   if(WIN32)
+      set_property(TARGET bson_shared APPEND PROPERTY SOURCES libbson.rc)
+   endif()
```
Comment (vector-of-bool) [inline src/libbson/CMakeLists.txt:235]:
```suggestion
   add_library(bson_shared SHARED $<$<PLATFORM_ID:Windows>:libbson.rc>)
   target_link_libraries(bson_shared PRIVATE $<BUILD_INTERFACE:bson_obj_pic>)
```

Same in libmongoc

**Review by vector-of-bool — APPROVED** [review]
One minor comment

**Pull request merged**